### ISSUE TITLE
used variables and arguments shows as unused in .js file

### ISSUE THREAD

**Issue by rpuls** [Microsoft/vscode#53591]
- VSCode Version: 1.24.1
- OS Version: Windows 10

Steps to Reproduce:

I can't give an exact step by step on how to reproduce this issue, in some larger .js files, the new feature that grey out unused variables will grey out variables and arguments that are used. When hover over the grey values the info box appears with following msg: 

> [ts] 'xxx' is declared but its' value is never read

Here is how it looks:
![alt text](https://image.ibb.co/eLLD7J/Capture.png "vscode")

If I copy the code and pastes it into a blank .js file, the issue does not apply. 

Does this issue occur when all extensions are disabled?: Yes

**Comment by mjbvz** [comment]
Does this reproduce in the [latest VS Code build](https://code.visualstudio.com/insiders/) with all extensions disabled?

**Comment by rpuls** [comment]
Yes, I just updated vscode to
Version: 1.25.0
Date: 2018-07-05T13:11:11.248Z

started vscode with `code . --disable-extensions` and confirmed that the "extensions are temporarily disabled" message appeared. The issue still applies.

I was not able to reproduce it in vs code insiders, as no variables showed as unused, even not when adding dummy var, const, let for debuggin. No grey out, and no "unused" flag on hover message.

**Comment by mjbvz** [comment]
Please share some example code (as text) or a small example project

**Comment by rpuls** [comment]
It is one particular file that has this problem. As I wanted to share the code I copied all its content into a new file I called sample.js - In the new file all variables and arguments looks fine, none of them are grey, and none of them has the "is never read" hover message. I have tried to add a unused variable to this particular document. The linter immediately notices that it is unused while vscode does not grey it, but greys other (this time with extensions enabled). I think it must be a bug in the "grey out feature". One more example from this document:
![alt text](https://image.ibb.co/jcCX98/Capture.png)
I'm sorry but I cannot share this project due to liability, and I have not yet reproduces the bug in any other files. Will create a small project and try to provoke the bug to appear.
